src/data_handling/timeseries.jl

```diff
@@ -48,6 +53,24 @@ function timeseries(notes, property, f, grid)
     return tvec, ts
 end
 
+"""
+    segment_notes(notes, grid)
+Quantizes the positions and durations of `notes` and segments them according to the duration of a grid unit.
+"""
+function segment_notes(notes, grid)
+    tpq = notes.tpq
+    segment_duration = Int(grid[2]*tpq)
+    grid_division = Int(tpq/segment_duration)
+    qnotes = quantize(notes, 0:1//grid_division:1)
+    segmented_notes = Notes()
+    for qn in qnotes
+        for segment_idx in 0:round(Int,qn.duration/segment_duration)-1
+            push!(segmented_notes, Note(qn.pitch, qn.velocity, qn.position + segment_idx*segment_duration, segment_duration))
+        end
+    end
+    return segmented_notes
```

Comment:
This should be replaced to `timesort!(segmented_notes)`, instead of `timesort!` used in the subsequent call.

Reply:
You mean sorting the notes in the 'segment_notes' function instead doing it in the 'timeseries' function ?

Reply:
Exactly.

Reply:
Done.